samples/cpp/live_detect_qrcode.cpp

```diff
@@ -0,0 +1,177 @@
+        "{o out    |        | output path to file (for mode: save_image) }";
+    CommandLineParser cmd_parser(argc, argv, keys);
+
+    cmd_parser.about("This program detects the QR-codes from camera or images using the OpenCV library.");
+    if (cmd_parser.has("help"))
+    {
+        cmd_parser.printMessage();
+        return 0;
+    }
+
+    string mode          = cmd_parser.get<string>("mode"); // detect mode
+    string in_file_name  = cmd_parser.get<string>("in");    // input  path to image
+    string out_file_name = cmd_parser.get<string>("out");   // output path to image
+
+    if (!cmd_parser.check())
+    {
+        cmd_parser.printErrors();
+        return -1;
+    }
+
+    int return_code = 0;
+    if (mode == "camera")
+    {
+        return_code = liveQRCodeDetect();
+    }
+    else if (mode.find("image") != string::npos)
+    {
+        return_code = showImageQRCodeDetect(mode, in_file_name, out_file_name);
+    }
+    else
+    {
+        cout << "Unsupported mode: " << mode << endl;
+        return_code = -1;
+    }
+    return return_code;
+}
+
+void getMatWithQRCodeContour(Mat &color_image, vector<Point> transform, int fps)
+{
+    if (!transform.empty())
+    {
+        double show_radius = (color_image.rows > color_image.cols)
+                    ? 3.375 * color_image.rows / color_image.cols
+                    : 3.375 * color_image.cols / color_image.rows;
+        double contour_radius = show_radius * 0.4;
+
+        vector< vector<Point> > contours;
+        contours.push_back(transform);
+        drawContours(color_image, contours, 0, Scalar(211, 0, 148), static_cast<int>(contour_radius));
+
+        circle(color_image, transform[0], static_cast<int>(show_radius), Scalar(255,   0,   0), -1);
+        circle(color_image, transform[1], static_cast<int>(show_radius), Scalar(  0, 255,   0), -1);
+        circle(color_image, transform[2], static_cast<int>(show_radius), Scalar(  0,   0, 255), -1);
+        circle(color_image, transform[3], static_cast<int>(show_radius), Scalar(255,   0, 255), -1);
+    }
+
+    ostringstream convert;
+    if (fps != -1) { convert << fps << " FPS."; }
+    else { convert << "Not find QR code"; }
+    putText(color_image, convert.str(), Point(25, 25), FONT_HERSHEY_DUPLEX, 1, Scalar(0, 0, 255), 2);
+}
+
+int liveQRCodeDetect()
+{
+    VideoCapture cap(0);
+    if(!cap.isOpened())
+    {
+        cout << "Not opened camera" << '\n';
+        return -4;
+    }
+
+    TickMeter total;
+    for(;;)
+    {
+        Mat frame, src;
+        vector<Point> transform;
+        cap >> frame;
+        if(frame.empty()) { break; }
+        cvtColor(frame, src, COLOR_BGR2GRAY);
+
+        total.start();
+        bool result_detection = detectQRCode(src, transform);
+        total.stop();
+        double fps = 1 / total.getTimeSec();
+        total.reset();
+        if (!result_detection) { fps = -1; }
+
+        getMatWithQRCodeContour(frame, transform, static_cast<int>(fps));
```

Comment:
1. use cvRound() instead of static_cast for better accuracy
2. do not pass fps to getMatWithQRCodeContour() function. Instead, render it right here. It's more logical. Visualization should be de-coupled from the processing part.